### PR TITLE
8927

### PR DESCRIPTION
**Pull request by tigerxn**
Fixes 8927.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/java/org/apache/shardingsphere/proxy/initializer/impl/AbstractBootstrapInitializer.java [hunk 1]
```diff
@@ -113,6 +113,8 @@ private void setDatabaseServerInfo() {
             DatabaseServerInfo databaseServerInfo = new DatabaseServerInfo(dataSourceSample.get());
             log.info(databaseServerInfo.toString());
             MySQLServerInfo.setServerVersion(databaseServerInfo.getDatabaseVersion());
+        } else {
+            MySQLServerInfo.setServerVersion(null);
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/java/org/apache/shardingsphere/proxy/initializer/impl/AbstractBootstrapInitializer.java:117]: set null maybe unnecessary

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/context/ProxyContext.java [hunk 2]
```diff
@@ -115,7 +116,13 @@ public ShardingSphereMetaData getMetaData(final String schemaName) {
...
             return Optional.empty();
         }
-        Map<String, DataSource> dataSources = getMetaData(schemaNames.get(0)).getResource().getDataSources();
-        return dataSources.values().stream().findFirst();
+        for (String schemaName : schemaNames) {
+            ShardingSphereResource shardingSphereResource = getMetaData(schemaName).getResource();
+            DatabaseType databaseType = shardingSphereResource.getDatabaseType();
+            if ("MySQL".equals(databaseType.getName())) {
+                return shardingSphereResource.getDataSources().values().stream().findFirst();
+            }
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/context/ProxyContext.java:124]: This is the public module, we cannot consider about MySQL only.
What is the behaviour of PostgreSQL?

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/java/org/apache/shardingsphere/proxy/initializer/impl/AbstractBootstrapInitializer.java (hunk 1 above)
Comment (tigerxn) [reply shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/main/java/org/apache/shardingsphere/proxy/initializer/impl/AbstractBootstrapInitializer.java:117]: OK

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/context/ProxyContext.java (hunk 2 above)
Comment (tigerxn) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/context/ProxyContext.java:124]: This method is only used by mysql, PostgreSQLdoes not set the version information.

Comment (tigerxn) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/context/ProxyContext.java:124]: The method(getDataSourceSample) name can be changed。

Comment (terrymanu) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/context/ProxyContext.java:124]: The ProxyContext is public class which is in shardingsphere-proxy-backend module.
The database dialect should in their own module, it is better to keep public module independent, and let them do not know database dialect details.

Comment (tigerxn) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/context/ProxyContext.java:124]: This method(getDataSourceSample) is moved from ProxyContext  to AbstractBootstrapInitializer ，and changed to private
method，only used by setDatabaseServerInfo。

Comment (tigerxn) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/context/ProxyContext.java]: please revivew again.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**